core/src/main/scala/org/apache/spark/status/api/v1/AllStagesResource.scala

```diff
@@ -59,6 +59,10 @@ private[v1] object AllStagesResource {
       stageUiData: StageUIData,
       includeDetails: Boolean): StageData = {
 
+    val firstTaskLaunchedTimeOption =
+      Option(stageUiData.taskData.values.map(_.taskInfo.launchTime).filter(_ > 0).min)
+    val firstTaskLaunchedTime = new Date(firstTaskLaunchedTimeOption.get)
```

Comment:
this isn't enough -- if the list of tasks with launchTime > 0 is empty, this will throw an exception:

```scala
scala> Option(Seq[Int]().min)
java.lang.UnsupportedOperationException: empty.min
```

also, I think the final type of `StageData#firstTaskLaunchTime` actually needs to be a `Option[Date]`, since it might be empty.